compiler/test/suites/pattern_matching.re

```diff
@@ -197,4 +203,49 @@ describe("pattern matching", ({test}) => {
     "match ((\"foo\", 5)) { (\"foo\", n) when n == 7 => false, (\"foo\", 9) when true => false, (\"foo\", n) when n == 5 => true, _ => false }",
   );
   assertFileRun("mixed_matching", "mixedPatternMatching", "true\n");
+  assertWarning(
+    "bool_exhaustiveness1",
+    "match (true) {
+       true => print(5),
+     }",
+    Warnings.PartialMatch("false"),
+  );
+  assertWarning(
+    "bool_exhaustiveness2",
+    "match (true) {
+       false => print(5),
+     }",
+    Warnings.PartialMatch("true"),
+  );
+  assertWarning(
+    "bool_exhaustiveness3",
+    "match (true) {
+       true => print(5),
+       true => print(5),
+     }",
+    Warnings.PartialMatch("false"),
+  );
+  assertNoWarning(
+    "bool_exhaustiveness4",
+    "match (true) {
+       false => print(5),
+       true => print(5),
+     }",
```

Comment:
Is there anything weird with `when` clause on a boolean constant? Can you add a test?

Reply:
Not really! Guarded clauses aren't considered when determining exhaustiveness, since the guard only makes the pattern match sometimes. You'll still get a "this isn't exhaustive" warning, but you get an extra message telling you that it's because that branch is guarded. Test incoming 🙂